src/lib.rs

```diff
@@ -520,6 +527,45 @@ where
     Ok(())
 }
 
+fn start_iter_ids_thread(
+    id_name: &str,
+    shard_num: u32,
+    moray_ip: String,
+    obj_tx: crossbeam_channel::Sender<SharkspotterMessage>,
+    log: Logger,
+    conf: config::Config,
+) -> impl Fn() -> () {
+    let moray_socket = format!("{}:{}", moray_ip, 2020);
+    let id_string = id_name.to_string();
+
+    move || {
+        if let Err(e) = iter_ids(
+            id_string.as_str(),
+            &moray_socket,
+            &conf,
+            log.clone(),
+            shard_num,
+            |value, shark, shard_num| {
+                let msg = SharkspotterMessage {
+                    value,
+                    shark: shark.to_string(),
+                    shard: shard_num,
+                };
+
+                obj_tx
+                    .send(msg)
+                    .map_err(|e| Error::new(ErrorKind::Other, e.description()))
+            },
+        ) {
+            error!(
+                &log,
+                "Encountered error scanning shard {} ({})", shard_num, e
+            );
+            // TODO panic if this is not an `_idx` or `_id` not found.
```

Comment:
is the TODO handled?

Reply:
I have a separate jira issue opened for that.  I need to think about the correct solution a bit more, and it different enough that I didn't want to include it in this change.